### ISSUE TITLE
Rfc8414 section-2.1 supported

### ISSUE THREAD

**Issue by cberger76** [DuendeArchive/IdentityModel#379]
I could not find a way to decode JWT signed discovery Documents according to

https://tools.ietf.org/html/rfc8414#section-2.1

or am i missing a point here ?

**Comment by leastprivilege** [comment]
There is no built-in support if that's what you mean. But you can get the `signed_metadata` element using `TryGetValue` and validate yourself.

**Comment by cberger76** [comment]
what is the correct entry point where i can decode the value myself? currently the GetDiscoveryDocumentAsync method fails with a exception (i think it assumes a valid json, but JWT is no valid json)

**Comment by leastprivilege** [comment]
The way I understand it is, that the discovery document is still a valid JSON object - but an additional `signed_metadata` element

> Signed metadata is included in the authorization server metadata JSON object using this OPTIONAL member

Maybe some other expected values are missing now because they are included in the signed element only.

Which server product is producing this? You also need to check the exact exception (including inner exception) to find out what's missing from the POV of IdentityModel.

**Comment by cberger76** [comment]
As much as i know it is a default JWT (i think it is a custom implementation used by Gematik https://github.com/gematik/ref-idp-server)

a JWT from their IDP looks like

`eyJhbGciOiJCUDI1NlIxIn0.eyJhdXRob3JpemF0aW9uX2VuZHBvaW50IjoiaHR0cDovL2Fwby1kb2NrZXItMDE6ODA4MC9hdXRob3JpemF0aW9uIiwidG9rZW5fZW5kcG9pbnQiOiJodHRwOi8vYXBvLWRvY2tlci0wMTo4MDgwL3Rva2VuIiwiaXNzdWVyIjoiaHR0cDovL2Fwby1kb2NrZXItMDE6ODA4MC9hdXRoL3JlYWxtcy9pZHAiLCJqd2tzX3VyaSI6Imh0dHA6Ly9hcG8tZG9ja2VyLTAxOjgwODAvandrcyIsImV4cCI6MTYxMzQ4NTE4MiwibmJmIjoxNjEzMzk4NzgyLCJpYXQiOjE2MTMzOTg3ODIsInB1a191cmlfYXV0aCI6Imh0dHA6Ly9hcG8tZG9ja2VyLTAxOjgwODAvYXV0aEtleS9qd2tzLmpzb24iLCJwdWtfdXJpX3Rva2VuIjoiaHR0cDovL2Fwby1kb2NrZXItMDE6ODA4MC90b2tlbktleS9qd2tzLmpzb24iLCJwdWtfdXJpX2Rpc2MiOiJodHRwOi8vYXBvLWRvY2tlci0wMTo4MDgwL2Rpc2NLZXkvandrcy5qc29uIiwic3ViamVjdF90eXBlc19zdXBwb3J0ZWQiOlsicGFpcndpc2UiXSwiaWRfdG9rZW5fc2lnbmluZ19hbGdfdmFsdWVzX3N1cHBvcnRlZCI6WyJCUDI1NlIxIl0sInJlc3BvbnNlX3R5cGVzX3N1cHBvcnRlZCI6WyJjb2RlIl0sInNjb3Blc19zdXBwb3J0ZWQiOlsib3BlbmlkIiwiZS1yZXplcHQiXSwicmVzcG9uc2VfbW9kZXNfc3VwcG9ydGVkIjpbInF1ZXJ5Il0sImdyYW50X3R5cGVzX3N1cHBvcnRlZCI6WyJhdXRob3JpemF0aW9uX2NvZGUiXSwiYWNyX3ZhbHVlc19zdXBwb3J0ZWQiOlsidXJuOmVpZGFzOmxvYTpoaWdoIl0sInRva2VuX2VuZHBvaW50X2F1dGhfbWV0aG9kc19zdXBwb3J0ZWQiOlsibm9uZSJdfQ.ZXfTHhKqi_w5z6JYH9r2Py78Mn9u2pA2AjpRQh2h8zxOlUvnGd9ytOdB5JSolMEdP0uF9w4DOh88s93DJajD1w`

when you put it to https://jwt.io/ you see they are split up in 3 parts like any JWT (but are no valid json)

if you extract the JWT the middle part (base64 encoded) contains the discovery document. The last part is the signature which can be verified against the keys from the JWKS uri.

so basically i can parse the JWT but i hoped i could plug into IdentityModel

**Comment by leastprivilege** [comment]
this is not spec compliant. sorry.

**Comment by cberger76** [comment]
ok thank you for the help, seems i was on the wrong track, i have to evaluate the implementation from the Gematik
